src/FSharp.Data.GraphQL.Server/Executor.fs

```diff
@@ -119,19 +115,20 @@ type Executor<'Root>(schema: ISchema<'Root>, middlewares : IExecutorMiddleware s
                           Errors = errors
                           FieldExecuteMap = fieldExecuteMap
                           Metadata = executionPlan.Metadata }
-                    let! res = runMiddlewares (fun x -> x.ExecuteOperationAsync) executionCtx executeOperation
+                    let! res = runMiddlewares (fun x -> x.ExecuteOperationAsync) executionCtx executeOperation |> AsyncVal.toAsync
                     return prepareOutput res
                 with
-                | ex -> return prepareOutput(GQLResponse.Error(ex.ToString(), executionPlan.Metadata))
+                | :? GraphQLException as ex -> return prepareOutput(GQLExecutionResult.Error(documentId, ex, executionPlan.Metadata))
+                | ex -> return prepareOutput(GQLExecutionResult.Error(documentId, ex.ToString(), executionPlan.Metadata)) // TODO: Handle better
```

Comment:
Please see: https://github.com/fsprojects/FSharp.Data.GraphQL/issues/425 (I just created this issue before looking at this pull request here ^^).

Reply:
Yes, the fix will come in that PR. So you will be able to do something like this.
```F#
// TODO: Handle your exceptions here and transform to IGQLError, optionally implement IGQLErrorExtensions for details
// GraphQLException already implements IGQLError
let parseError (ex : exn) =
    match ex with
    | :? ResolverErrorException as err ->
       [err :> IGQLError]
    | :? ResolverErrorExceptions as errs ->
      errs.InnerExceptions |> Seq.cast<IGQLError> |> Seq.toList
    | _  ->
        [{ new IGQLError with member _.Message = ex.GetType().FullName
            interface IGQLErrorExtensions with
            member error.Extensions =
                seq {
                    // TODO: Do not expose in production
                    kvp "stackTrace" (box ex.StackTrace)
                    match ex with
                    | :? AggregateException as agg ->
                       kvp "innerExceptions" (box agg.InnerExceptions)
                    | _ -> ()
                }
                |> ImmutableDictionary.CreateRange<string, obj> :> IReadOnlyDictionary<string, obj>
                |> ValueSome
            }]

let schemaConfig =
    { SchemaConfig.Default with ParseError = parseError }
```

Reply:
great! :+1: